resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
mathematical accident... ok.

You code
```
    @Test
    public void test() {
        for (int i = -10; i <= 0; i++) {
            System.out.print(toIndex(i, 5) + " ");
        }
    }

    protected static int toIndex(int anyInt, int len) {
        int index = anyInt % len;
        if (anyInt < 0) { // anyInt instead of index
            index = index + len - 1;
        }
        return index;
    }
```
4 0 1 2 3 4 0 1 2 **3 0**

Mine:

```
    @Test
    public void test() {
        for (int i = -10; i < 0; i++) {
            System.out.print(toIndex(i, 5) + " ");
        }
    }

    protected static int toIndex(int anyInt, int len) {
        int index = anyInt % len;
        if (index < 0) {
            index = index + len;
        }
        return index;
    }
```
0 1 2 3 4 0 1 2 3 4

If we are talking about "continuity". I think, we should close this discussion, submitted code produces expected output.